### PR TITLE
Make upsert modules

### PR DESCRIPTION
**Pull request by thallesfreitaszup**
## Issue
When a deploy is triggered and comes from a module that was updated on moove an error occurs.
```
{"requestId":"a4681a70-b0d5-11ea-94d1-01341d801058","timestamp":"2020-06-17T20:03:42.244Z","level":"error","message":""Could not find any entity of type "ComponentEntity" matching: {\n    "id": "25b9388e-6e6f-4dd8-87ca-c9f08698ef1c"\n}""}
```
## Solution

Perform a upsert on module when deploy is triggered

### CODE REVIEW

**Comment by thallesfreitaszup** [comment]
[Realizar upsert na hora de criar módulo no butler. Hoje, caso o módulo seja atualizado no moove, um erro é exibido.](https://trello.com/c/vlbLe2iR/538-realizar-upsert-na-hora-de-criar-m%C3%B3dulo-no-butler-hoje-caso-o-m%C3%B3dulo-seja-atualizado-no-moove-um-erro-%C3%A9-exibido)

**Comment by thallesfreitaszup** [comment]
[Realizar upsert na hora de criar módulo no butler. Hoje, caso o módulo seja atualizado no moove, um erro é exibido.](https://trello.com/c/vlbLe2iR/538-realizar-upsert-na-hora-de-criar-m%C3%B3dulo-no-butler-hoje-caso-o-m%C3%B3dulo-seja-atualizado-no-moove-um-erro-%C3%A9-exibido)

File: butler/src/app/api/deployments/services/modules.service.ts [hunk 1]
```diff
@@ -39,13 +39,12 @@ export class ModulesService {
   private async saveModule(moduleEntity: ModuleEntity) {
     const module = await this.moduleEntityRepository.findOne({ id: moduleEntity.id })
     const newComponents: ComponentEntity[] = moduleEntity.components.filter(
-      component => !module?.components.includes(component)
```
Comment (cpgo) [inline butler/src/app/api/deployments/services/modules.service.ts]: We should also fix the tests now that we know that the test was passing with a bug on the code.
That means the test was not accurate.

File: butler/src/app/api/deployments/services/modules.service.ts [hunk 2]
```diff
@@ -37,12 +40,35 @@ export class ModulesService {
...
+    }
+
+  }
+
+  private async updateAndSaveComponent(newComponent: ComponentEntity, module: ModuleEntity) {
+    const oldComponent = module.components.find(
+      component => component.name === newComponent.name
+    )
+    if(oldComponent){
+      newComponent.pipelineOptions = oldComponent.pipelineOptions
```
Comment (lucasbfernandes) [inline butler/src/app/api/deployments/services/modules.service.ts:67]: I don't think whe should do this

File: butler/src/app/api/deployments/services/modules.service.ts [hunk 3]
```diff
@@ -37,12 +40,35 @@ export class ModulesService {
...
+      await this.componentEntityRepository.delete(oldComponent.id)
```
Comment (lucasbfernandes) [inline butler/src/app/api/deployments/services/modules.service.ts:68]: Components shouldn't be deleted. If we do this, we will lose the entire cluster state

File: butler/src/app/api/deployments/services/modules.service.ts (hunk 1 above)
Comment (thallesfreitaszup) [reply butler/src/app/api/deployments/services/modules.service.ts]: fixed unit tests

File: butler/src/app/api/deployments/dto/create-deployment/create-component-deployment.dto.ts [hunk 4]
```diff
@@ -47,6 +47,6 @@ export class CreateComponentDeploymentDto {
   }
 
   public toComponentEntity(): ComponentEntity {
-    return new ComponentEntity(this.componentId)
+    return new ComponentEntity(this.componentId,this.componentName)
```
[the remark this answers is not in the document]
Comment (thallesfreitaszup) [reply butler/src/app/api/deployments/dto/create-deployment/create-component-deployment.dto.ts]: fixed

**Pull request merged**